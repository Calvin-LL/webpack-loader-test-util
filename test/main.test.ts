import crypto from "crypto";

import MyWebpackTestCompiler from "./helpers/MyWebpackTestCompiler";

describe.each([4, 5] as const)("main", (webpackVersion) => {
  it("should compile with basic options", async () => {
    const compiler = new MyWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({});

    const pictureBuffer = await bundle.readAssetAsPNG(
      "Macaca_nigra_self-portrait_large.jpg"
    );
    const pictureHash = crypto
      .createHash("sha256")
      .update(pictureBuffer)
      .digest("hex");

    expect(bundle.execute("main.js")).toMatchSnapshot();
    expect(pictureHash).toMatchSnapshot();
  });

  it("should compile with file content override", async () => {
    const compiler = new MyWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({ fileContent: "__export__ = 3" });

    expect(bundle.execute("main.js")).toMatchSnapshot();
  });

  it("should compile with specific file", async () => {
    const compiler = new MyWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({ entryFileName: "anotherIndex.js" });

    expect(bundle.execute("main.js")).toMatchSnapshot();
  });
});
