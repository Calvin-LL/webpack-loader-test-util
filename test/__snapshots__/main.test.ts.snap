// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should compile with basic options 1`] = `
Object {
  "default": "Macaca_nigra_self-portrait_large.jpg",
}
`;

exports[`main should compile with basic options 2`] = `"4321feb710c57ad8b5e0259f4491cc7e7a80d9a41658f5431039bc393f964de0"`;

exports[`main should compile with basic options 3`] = `
Object {
  "default": "Macaca_nigra_self-portrait_large.jpg",
}
`;

exports[`main should compile with basic options 4`] = `"4321feb710c57ad8b5e0259f4491cc7e7a80d9a41658f5431039bc393f964de0"`;

exports[`main should compile with file content override 1`] = `3`;

exports[`main should compile with file content override 2`] = `3`;

exports[`main should compile with specific file 1`] = `
Object {
  "default": "good",
}
`;

exports[`main should compile with specific file 2`] = `
Object {
  "default": "good",
}
`;
